v3.0 rebase branch

bjarki-andreasen opened the pull request:
v3.0 rebase branch

rlubos reviewed (changes requested):
Or actually, there are a few `nrf-squash!` commits, those should be cleaned up during the rebase.

bjarki-andreasen commented:
> Or actually, there are a few `nrf-squash!` commits, those should be cleaned up during the rebase.

Removed a Revert commit pair and squashed the three nrf-squash! commits
